enterprise/api_client/ecommerce.py

```diff
@@ -65,3 +70,31 @@ def get_course_final_price(self, mode, currency='$', enterprise_catalog_uuid=Non
         if price != mode['min_price']:
             return format_price(price, currency)
         return mode['original_price']
+
+    def create_ecommerce_order_for_manual_course_enrollment(  # pylint: disable=invalid-name
+            self,
+            enrolled_learner_lms_user_id,
+            enrolled_learner_username,
+            enrolled_learner_email,
+            enrolled_course_run_key,
+    ):
+        """
+        Create an ecommerce order record for manually enrolled learner.
+        """
+        try:
+            return self.client.manual_course_enrollment_order.post({
+                'lms_user_id': enrolled_learner_lms_user_id,
+                'username': enrolled_learner_username,
+                'email': enrolled_learner_email,
+                'course_run_key': enrolled_course_run_key,
+            })
+        except (SlumberBaseException, ConnectionError, Timeout) as exc:
+            LOGGER.exception(
+                '[Enterprise Manual Course Enrollment] Failed to create ecommerce order. LMSUserId: %s, User: %s, '
+                'Email: %s, Course: %s, Message: %s',
+                enrolled_learner_lms_user_id,
+                enrolled_learner_username,
+                enrolled_learner_email,
+                enrolled_course_run_key,
+                exc
+            )
```

Comment:
do you intend to swallow this exception or have it bubbled up to the caller? if it's the latter (which i would recommend), then you will have to add `raise exc` here

Reply:
If we do `raise exc` then we have to catch all the exceptions in the caller that we are already handling here. And in the caller we would catch the exceptions and return a `False` in the `except` block. I am not sure what benefit bubbling up the exception will give us in this case?

Reply:
On second thought, bubbling up exception will help in other cases, for example, when we do enrollment using enrollment api. So, I have created a custom exception and raised that exception.

Reply:
generally speaking, i think it is wrong to catch exceptions like this in the api client code because it typically masks problems when they occur. we should only "swallow" exceptions when the consequence can be easily handled for the business logic, ie trying to return a specific field and finding a `KeyError` or the like. only the caller can decide the consequence of getting an exception like this, and if it doesn't catch it then at least our logs and new relic will be loud about something being wrong.

Reply:
hmm...so in the client method(`create_order_for_manual_course_enrollment`), just re-raise exception and then let the caller of client method decide what to do with raised exceptions, right?